benchmark_test.go

```diff
@@ -218,3 +220,87 @@ func TestBenchmarkStatsPerPeer(t *testing.T) {
 	})
 	assert.Equal(t, want, statsServer.Aggregated(), "unexpected stats")
 }
+
+func TestBenchmarkOutput(t *testing.T) {
+	// Testing both plaintext and JSON output
+	tests := []struct {
+		name          string
+		format        []string
+		wantJSON      bool
+		wantOutput    []string
+		notWantOutput []string
+	}{
+		{
+			name:          "json",
+			format:        []string{"json", "JSON", "Json", "jsON", "jsoN"},
+			wantJSON:      true,
+			wantOutput:    []string{"summary", "benchmarkParameters", "maxRPS", "latencies"},
+			notWantOutput: []string{"Errors", "Benchmark parameters", "Max RPS", "Unrecognized format option"},
+		},
+		{
+			name:          "unspecified",
+			format:        []string{""},
+			wantJSON:      false,
+			wantOutput:    []string{"Benchmark parameters", "Max RPS"},
+			notWantOutput: []string{"Errors", "summary", "maxRPS", "Unrecognized format option"},
+		},
+		// Unimplemented format should default to plaintext
+		{
+			name:          "unrecognized",
+			format:        []string{"csv", "plaintext", "blob"},
+			wantJSON:      false,
+			wantOutput:    []string{"Benchmark parameters", "Max RPS", "Unrecognized format option"},
+			notWantOutput: []string{"Errors", "benchmarkParameters", "maxRPS"},
+		},
+	}
+
+	var requests atomic.Int32
+	s := newServer(t)
+	defer s.shutdown()
+	s.register(fooMethod, methods.errorIf(func() bool {
+		requests.Inc()
+		return false
+	}))
+	m := benchmarkMethodForTest(t, fooMethod, transport.TChannel)
+
+	for _, tt := range tests {
+		for _, format := range tt.format {
+			t.Run(tt.name, func(t *testing.T) {
+				requests.Store(0)
+				buf, _, out := getOutput(t)
+				opts := Options{
+					BOpts: BenchmarkOptions{
+						MaxRequests: 10,
+						MaxDuration: 200 * time.Millisecond,
+						RPS:         120,
+						Connections: 50,
+						Concurrency: 2,
+						Format:      format,
```

Comment:
since we want a really quick benchmark (we just want to validate the output rather than the benchmark run), suggest tweaking the parameters to make it run even quicker:
 - max requests is fine, but could drop to 1
 - max duration of like 100ms
 - no RPS specified (means run as fast as possible)
 - warmup = 0
 - Connections = 1, concurrency = max requests (to make sure they all start at once)